File: wiki_examples/src/main/java/com/github/klee0kai/wiki/init/StarsDependencies2.java
package com.github.klee0kai.wiki.init;

import com.github.klee0kai.stone.annotations.module.Module;
import com.github.klee0kai.stone.annotations.module.Provide;
import com.github.klee0kai.test.mowgli.galaxy.Sun;

@Module
public interface StarsDependencies2 {

    @Provide(cache = Provide.CacheType.Factory)
    Sun sun();

}

File: wiki_kotlin/src/main/kotlin/com/github/klee0kai/wiki/init/StarsDependencies2.kt
package com.github.klee0kai.wiki.init

import com.github.klee0kai.stone.annotations.module.Module
import com.github.klee0kai.stone.annotations.module.Provide
import com.github.klee0kai.test.mowgli.galaxy.Sun

@Module
interface StarsDependencies2 {

    @Provide(cache = Provide.CacheType.Factory)
    fun sun(): Sun

}
